lib/services/watcher.go

```diff
@@ -1009,28 +1065,23 @@ func (c *caCollector) resourceKind() string {
 
 // getResourcesAndUpdateCurrent refreshes the list of current resources.
 func (c *caCollector) getResourcesAndUpdateCurrent(ctx context.Context) error {
-	updatedCerts := make(CertAuthorityTypeMap)
+	var cas []types.CertAuthority
 
-	for _, caType := range c.WatchCertTypes {
-		cas, err := c.AuthorityGetter.GetCertAuthorities(ctx, caType, false)
+	for _, t := range c.Types {
+		authorities, err := c.AuthorityGetter.GetCertAuthorities(ctx, t, false)
 		if err != nil {
 			return trace.Wrap(err)
 		}
 
-		updatedCerts[caType] = make(CertAuthorityMap, len(cas))
-		for _, ca := range cas {
-			updatedCerts[caType][ca.GetName()] = ca
-		}
+		cas = append(cas, authorities...)
 	}
 
 	c.lock.Lock()
-	c.collectedCAs = updatedCerts
-	c.lock.Unlock()
+	defer c.lock.Unlock()
 
-	select {
-	case <-ctx.Done():
-		return trace.Wrap(ctx.Err())
-	case c.CertAuthorityC <- updatedCerts.ToSlice():
+	for _, ca := range cas {
+		c.cas[ca.GetType()][ca.GetName()] = ca
```

Comment:
Do you know if `GetCertAuthorities` is guaranteed to doublecheck the type of the CAs? If not we should probably doublecheck it here.